New block Loader

Naaaaiix opened the pull request:
# Description

This branch deprecates the `lazy` and `suspense` implementation and replaces it with a block loader. What do we expect of this implementation:
1. It's not an optimization, so things aren't faster. Its aim is to create the illusion that Maxi is faster.
2. Improves the UX (user experience): instead of spending time in front of an empty blank page that can even return some waiting advice (`the page doesn't respond, close or wait`), it makes Maxi blocks wait for the editor to be loaded, and then it loads the blocks. 
3. It doesn't make the load longer or slower: after many comparisons, we saw the average time between this branch and a branch without this implementation is the same. To test it by your own, you can use [this branch](https://github.com/maxi-blocks/maxi-blocks/tree/revert-suspense) (called `revert-suspense`). But the main point is: we offer some movements to creators that see changes on the screen while everything is loading, which at the end is much better than an empty page.
4. It just works when we load the page and there are saved Maxi blocks. So, in new pages or when downloading from the Library, this loader is disabled. Just works when loading a page that already has blocks; once loaded the content, the loader is disabled.

# How Has This Been Tested?

The first test is to compare metrics. Here's a video between this branch and `revert-suspense` branch: 

https://user-images.githubusercontent.com/50477222/225551497-9cecd2fc-1400-4205-8302-a46202c15ae0.mp4

(This video was recorded for @kyrapieterse and @ckp267, so it's out of context)

Then, other things checked:
1. The loader just works when the page already contains Maxi blocks
2. On a new page, the loader is not appearing when adding blocks
3. When adding from the Library or pasting code, the loader doesn't appear

# Test checklist

<!--- Please remove the unnecessary checkbox --->

**_ Front/Back Testing _**

-   [ ] Test the aspects commented above

**_ Pre-Code Testing _**

-   [ ] Test the aspects commented above
-   [ ] Check no commented code and no unnecessary imports
-   [ ] Standards of the project have been followed
-   [ ] No errors/warnings on console

# Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] My changes generate no new warnings/errors
-   [ ] I have added/updated tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes

Naaaaiix committed:
Remove lazy load
Improved withSuspense lifecycle

Naaaaiix committed:
Fix tests

Naaaaiix committed:
Remove console.log

Naaaaiix committed:
Implement lighter Suspense

Naaaaiix committed:
Fixes `withMaxiSuspense` just for the page load

Naaaaiix committed:
Merge branch 'master' into lighter-suspense

Naaaaiix committed:
Fix package-lock.json

Naaaaiix committed:
Replace the name of the loader HOC

Marko64 reviewed (commented):
I like this better. But one thing can get on one's nerves:
https://user-images.githubusercontent.com/19425503/225557739-b2b72e51-7089-4439-ad3e-dfd7625aacf1.mp4

Naaaaiix committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks

elzadj reviewed (approved):
Bye bye suspense

Naaaaiix committed:
Main Maxi block component optimization (#4558)

* Revert Suspense implementation

* Abstract RowBlockTemplate component
Reduce withMaxiProps load
Reduce MaxiBlockComponent load
Abstracted some MaxiBlockComponent methods
Avoid rendering Inspector if it's not selected

* Clean Text Maxi main file

* Clean code

* Library insert optimization (#4559)

* Refactored `onRequestInsertPattern`
Modified Library flow when clicking insert

* Fix package-lock.json

* Fix CORS
Clean code

* Fixes placeholder source

Naaaaiix committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks

Naaaaiix committed:
Merge branch 'master' into lighter-suspense

Naaaaiix committed:
Fix package.json

Naaaaiix committed:
Fix typo

Naaaaiix committed:
Fix tests

Naaaaiix committed:
Fix tests

Naaaaiix committed:
Fixes compatibility with React 18
